src/trace-plugin-loader.js

```diff
@@ -49,6 +49,19 @@ function checkLoadedModules() {
   }
 }
 
+function checkPatch(patch) {
+  if (!patch.patch && !patch.intercept) {
+    throw new Error('Plugin patch for ' + patch + ' doesn\'t patch ' +
+      'anything.');
+  } else if (patch.patch && patch.intercept) {
+    throw new Error('Plugin patch for ' + patch.file + ' has ' +
+      'both intercept and patch functions.');
+  } else if (patch.unpatch && patch.intercept) {
```

Comment:
I think having a warnings for patch without unpatch and unpatch without patch might be good. I think with these combinations a separate warning for unpatch with intercept is obsolete.